includes/api/class-edd-api.php

```diff
@@ -1245,72 +1248,96 @@ public function get_stats( $args = array() ) {
 					$start_date = $dates['year'] . '-' . $dates['m_start'] . '-' . $dates['day_start'];
 					$end_date   = $dates['year_end'] . '-' . $dates['m_end'] . '-' . $dates['day_end'];
 
-					$stats = EDD()->payment_stats->get_sales_by_range( 'other', true, $start_date, $end_date );
+					// Force the data for the reports API.
+					$_GET['filter_from'] = $start_date;
+					$_GET['filter_to']   = $end_date;
+					$_GET['range']       = 'other';
 
-					foreach ( $stats as $sale ) {
-						$key                    = $sale['y'] . $sale['m'] . $sale['d'];
-						$sales['sales'][ $key ] = (int) $sale['count'];
-					}
+					$stats   = new EDD\Stats();
+					$dates   = EDD\Reports\parse_dates_for_range();
+
+					$total_sales = $stats->get_order_count(
+						array(
+							'start'        => $dates['start']->format( 'Y-m-d H:i:s' ),
+							'end'          => $dates['end']->format( 'Y-m-d H:i:s' ),
+							'revenue_type' => 'net',
+						)
+					);
 
-					$start_date = date( 'Y-m-d', strtotime( $start_date ) );
-					$end_date   = date( 'Y-m-d', strtotime( $end_date ) );
+					$start_date  = $dates['start']->format( 'Y-m-d' );
+					$end_date    = $dates['end']->format( 'Y-m-d' );
 
 					while ( strtotime( $start_date ) <= strtotime( $end_date ) ) {
+
+						// Force the data for the reports API.
+						$_GET['filter_from'] = $start_date;
+						$_GET['filter_to']   = $start_date;
+						$_GET['range']       = 'other';
+
+						$dates = EDD\Reports\parse_dates_for_range();
+
 						$d = date( 'd', strtotime( $start_date ) );
 						$m = date( 'm', strtotime( $start_date ) );
 						$y = date( 'Y', strtotime( $start_date ) );
 
 						$key = $y . $m . $d;
 
 						if ( ! isset( $sales['sales'][ $key ] ) ) {
-							$sales['sales'][ $key ] = 0;
+							$sales['sales'][ $key ] = $stats->get_order_count(
+								array(
+									'start'        => $dates['start']->format( 'Y-m-d H:i:s' ),
+									'end'          => $dates['end']->format( 'Y-m-d H:i:s' ),
+									'revenue_type' => 'net',
```

Comment:
If line 1256 is updated then I think this can be removed, as it's the same instance of the class? But noting that setting the `revenue_type` here may not currently work as expected.